### DISCUSSION TITLE
Option to have the popup open -22px vertically relative to the current cursor position

### DISCUSSION THREAD

**Discussion by boettges** [will-stone/browserosaurus#506] [Ideas]
Hello,

First of all, thanks a lot for creating this great little helper. It's a joy to use ☺️

I have just one small suggestion to improve the ease of use: 
When the popup opens, it seems positioned 1px below the current cursor's position. That includes the window's cast shadow at the top. The upside is that just another click will dismiss the popup. Yet, cursor movement is necessary to select the default browser. 

The popup is already shifted by (if I'm not mistaken) -70px horizontally by default. Meaning there is already a logic implemented to avoid edge-case collisions (screen edges, multi-display junction areas, etc.).

It would be great to have an option to use this relative positioning by -22px vertically. That way, the opening popup is slightly shifted to the top so that the cursor is positioned just above the default browser button. That allows performing two consecutive clicks to open the link in the default browser without any repositioning of the mouse or leaving the trackpad to hit space to select. 

Sure, nitpicky details, but isn't that what Browsersaurus is about anyways? 😅 I would love to hear your thoughts on this.

Thanks a bunch!

**Comment by will-stone** [comment]
Hi @boettges 👋 You hit the nail on the head here:

> The upside is that just another click will dismiss the popup.

This was to avoid accidental browser opening when all you want to do is check the underlying URL. This is why the favourite browser is so large; it means [Fitts's law](https://en.wikipedia.org/wiki/Fitts%27s_law) is satisfied, whilst also guarding against accidental double-clicks.

There's also muscle memory to think of: after a while, all you're going to do is double-click to open links, making Browserosaurus moot. You'll do it even when you meant to open other browsers. By it being two different actions, click + move-click or click + keyboard, you make it more deliberate and thought about, and less likely to simply open the favourite when you meant to open a lesser-used browser in this instance.

**Reply by boettges** [reply]
I see the point, of course. Yet, that is why I thought providing it as a non-default option in the preferences would come in handy for some users.

I had the same hesitation before, but as a proof of concept, I have been using the automation tool KeyboardMaestro to move the popup by said pixel count vertically for some time now. There aren't more accidental activations (if any) than with the default behavior. It also took no time to adapt and feels more natural due to the removed obstacle.

As a user of modern computer systems, the dreadful double-click to open a link is long gone. That is why I specifically called it "two consecutive clicks" instead of a double-click. Browserosaurus will still have its intercepting nature and introduce a brief pause opposing a strict zero delay double-click. Yet, despite the interruption, the interaction feels much quicker due to the reduced friction in the process. 

It comes down to how many browser-external links one opens per day and how the split is amongst the various browsers. In my case, I open the default browsers about 90% of the time.

**Reply by will-stone** [reply]
Hey @boettges, you'll pleased to know that the next major version of Browserosaurus will be trialling having the favourite browser under the mouse on load 🙂 I think my original reason for not moving it is no longer holding. We'll see how it goes. No ETA yet but should be soon.
